src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<E extends Machine> implements MachineService<E> {
+    @Override
+    public List<Machine> getAll(Class<? extends E> type) {
+        if (Bulldozer.class.equals(type)) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            return new ArrayList<>(bulldozerProducer.get());
```

Comment:
you can avoid creating object in 17 line and just
```suggestion
            return new ArrayList<>(new BulldozerProducer().get());
```